lib/auth/user_gateway.dart

```diff
@@ -50,24 +56,36 @@ class UserGateway {
 
 class User {
   final String id;
-  final String displayName;
-  final String photoUrl;
   final String email;
   final bool emailVerified;
+  final String displayName;
+  final String photoUrl;
+  final String phoneNumber;
+  final bool disabled;
+  final DateTime tokensValidAfterTime;
 
   User.fromMap(Map<String, dynamic> map)
       : id = map['localId'],
         displayName = map['displayName'],
         photoUrl = map['photoUrl'],
         email = map['email'],
-        emailVerified = map['emailVerified'];
+        emailVerified = map['emailVerified'],
+        phoneNumber = map['phoneNumber'],
+        disabled = map['disabled'] ?? false,
+        tokensValidAfterTime = map['validSince'] == null
+            ? null
+            : DateTime.fromMillisecondsSinceEpoch(
+                int.parse(map['validSince']) * 1000);
```

Comment:
What is the validSince field? Can you link to the documentation?

Reply:
firebase_admin :grimacing:

Reply:
Please understand what the code does before importing it into this project.

Reply:
This is self explanatory though. I've tested firebase_admin before, it works perfectly, therefore validSince must work correctly as well.
https://cloud.google.com/identity-platform/docs/reference/rest/v1/projects.accounts/lookup
This is the doc on the return from the lookup post request. It doesn't mention it. But since firebase_admin works, then the docs surely must be out of date.

Reply:
But we don't know what that field means since it's not in the docs. I don't think it's self explanatory at all.
It could just as well be a deprecated field, which is why it's not in the documentation.

Reply:
I am now getting conflicting information. firebase_admin uses auth_time, while the firebase java sdk uses `iat` (issued_at) https://firebase.google.com/docs/reference/admin/node/admin.auth.DecodedIdToken#iat

![image](https://user-images.githubusercontent.com/7407478/82753176-5ef19000-9dcc-11ea-8e2a-a9ea50683701.png)

Reply:
java sdk has validSince 
![image](https://user-images.githubusercontent.com/7407478/82753349-89901880-9dcd-11ea-99f9-50410493a057.png)

Reply:
Seriously, please stop basing your implementation so strongly on other projects. You should understand what code does before implementing it, especially for something as critical as token validation.

Reply:
The "other products" are the officialy supported products that are open source. The documentation does not give me enough liberty to implement this blindly, I have to rely on them.

firebase_admin and the java firebase admin sdk both match implementation. The doc literally doesn't mention `validSince` anywhere.

I specifically said I wanted your help with this whole thing, I mentioned in the issue I made that I don't have a strong enough understanding about all of this to make it myself. You suggested I make a PR anyways, because if I don't, no one will. 

Please be patient and bear with me here, I need your help on this.

Reply:
I already told you what you need to do. This implementation should've been a two-liner and done.

My problem with basing your implementation on other projects is that they have a different architecture and goals.

Regarding validSince it's a part of the JWT token payload that's already being verified in jwt.dart. In our code it's called issueTime, which is the standard naming convention for that field, and why this was so confusing.

Reply:
@cachapa No no, the jwt token's payload has `issued time` ("iat") which is different from validSince.
The jwt token uses this: https://firebase.google.com/docs/reference/admin/node/admin.auth.DecodedIdToken
```
The ID token's issued-at time, in seconds since the Unix epoch. That is, the time at which this ID token was issued and should start to be considered valid.

The Firebase SDKs transparently refresh ID tokens every hour, issuing a new ID token with a new issued-at time. If you want to get the time at which the user session corresponding to the ID token initially occurred, see the auth_time property.
```

While the user object or user record uses validSince.
https://firebase.google.com/docs/reference/rest/auth
```
The timestamp, in seconds, which marks a boundary, before which Firebase ID token are considered revoked.
```

Yes I just found this. Yes it took me a while, yes I'm sorry.